stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/feature/login/LoginFragment.kt

```diff
@@ -6,63 +6,98 @@ import androidx.core.view.isVisible
+        }
+
         viewModel.state.observe(
             viewLifecycleOwner,
             Observer {
                 when (it) {
-                    is State.AvailableUsers -> renderAvailableUsers(it.availableUsers)
                     is State.RedirectToChannels -> redirectToChannelsScreen()
-                    is State.Loading -> changeLoadingIndicatorVisibility(true)
-                    is State.Error -> showErrorMessage(it.errorMessage)
                     is State.RedirectToChannel -> redirectToChannel(it.cid)
+                    is State.Loading -> showLoading()
+                    is State.Error -> showErrorMessage(it.errorMessage)
+                    is State.ValidationError -> showValidationErrors(it.invalidFields)
                 }
             }
         )
+
         activity?.intent?.getStringExtra(EXTRA_CHANNEL_ID)?.let { cid ->
             viewModel.targetChannelDataReceived(cid)
         }
     }
 
-    private fun redirectToChannel(cid: String) {
-        findNavController().navigateSafely(LoginFragmentDirections.actionLoginFragmentToChannelFragment(cid))
+    private fun collectCredentials(): LoginCredentials {
+        return LoginCredentials(
+            apiKey = apiKeyEditText.text.toString().trim(),
+            userId = userIdEditText.text.toString().trim(),
+            userToken = userTokenEditText.text.toString().trim(),
+            userName = userNameEditText.text.toString().trim()
+        )
     }
 
-    private fun redirectToChannelsScreen() {
-        findNavController().navigateSafely(R.id.action_loginFragment_to_channelsFragment)
+    private fun showLoading() {
+        changeLoadingIndicatorVisibility(true)
+        clearValidationErrors()
     }
 
-    private fun renderAvailableUsers(users: List<User>) {
+    private fun showErrorMessage(errorMessage: String?) {
         changeLoadingIndicatorVisibility(false)
-        adapter.setUsers(users)
+        showToast(errorMessage ?: getString(R.string.backend_error_info))
     }
 
-    private fun showErrorMessage(errorMessage: String?) {
+    private fun showValidationErrors(invalidFields: List<ValidatedField>) {
         changeLoadingIndicatorVisibility(false)
-        showToast(
-            errorMessage ?: getString(R.string.backend_error_info)
-        )
+
+        invalidFields.forEach {
+            when (it) {
+                ValidatedField.API_KEY -> apiKeyEditText
+                ValidatedField.USER_ID -> userIdEditText
+                ValidatedField.USER_TOKEN -> userTokenEditText
+            }.error = getString(R.string.login_validation_error)
+        }
+    }
+
+    private fun clearValidationErrors() {
+        apiKeyEditText.error = null
+        userIdEditText.error = null
+        userTokenEditText.error = null
     }
 
     private fun changeLoadingIndicatorVisibility(isVisible: Boolean) {
         loadingProgressBar.isVisible = isVisible
-        usersList.isVisible = !isVisible
+    }
```

Comment:
At this point, we might as well remove this method, it's body is a single line that's shorter than actually calling it